python/tvm/driver/tvmc/workspace_pools.py

```diff
@@ -0,0 +1,199 @@
+                Example usage: --workspace-pools-write-latency-cycles=flash:8""",
+    )
+    parser.add_argument(
+        "--workspace-pools-target-burst-bytes",
+        help="""The burst length of the memory pool in bytes per target.
+                Example usage: --workspace-pools-target-burst-bytes=flash:accel:1""",
+    )
+
+
+def _parse_target_burst(attr, pool_name):
+    if pool_name not in attr:
+        return {}
+
+    return {target: int(attr[pool_name][target]) for target in attr[pool_name]}
+
+
+def _parse_target_access(attr, pool_name):
+    try:
+        return attr[pool_name]
+    except:
+        raise TVMCException(f'Workspace Pool "{pool_name}" using undefined Target.')
+
+
+def _split_pools(attr):
+    return re.split(",", attr) if attr else []
+
+
+def _target_attributes_to_pools(attr, targets):
+    if not targets or attr is None:
+        return {}
+
+    target_attributes = {}
+    for pool_values in re.split(",", attr):
+        pool_name, target_name, target_value = re.split(":", pool_values)
+        if pool_name not in target_attributes:
+            target_attributes[pool_name] = {}
+
+        matched_targets = [target for target in targets if target_name == target.kind.name]
+        if matched_targets:
+            target_attributes[pool_name][matched_targets[0]] = target_value
+        else:
+            raise TVMCException(
+                "The workspace pool target specification for target access "
+                "needs to be the same TVM target as when specifying targets to use."
+            )
+
+    return target_attributes
+
+
+def _attribute_to_pools(attr):
+    return dict(pool.split(":", maxsplit=1) for pool in re.split(",", attr)) if attr else {}
+
+
+def workspace_pools_recombobulate(parsed, targets):
+    """Reconstructs the Workspace Pools args and returns a WorkspaceMemoryPool object"""
+    WORKSPACE_POOL_PARAMS = [
+        "workspace_pools_size_hint_bytes",
+        "workspace_pools_target_access",
+        "workspace_pools_clock_frequency_hz",
+        "workspace_pools_read_bandwidth_bytes_per_cycle",
+        "workspace_pools_write_bandwidth_bytes_per_cycle",
+        "workspace_pools_read_latency_cycles",
+        "workspace_pools_write_latency_cycles",
+        "workspace_pools_target_burst_bytes",
+    ]
+    WORKSPACE_POOL_TARGET_PARAMS = [
+        "workspace_pools_target_access",
+        "workspace_pools_target_burst_bytes",
+    ]
+
+    workspace_pools = _split_pools(parsed.workspace_pools)
+    pool_to_attributes = {
+        workspace_pool_param: _attribute_to_pools(getattr(parsed, workspace_pool_param))
+        for workspace_pool_param in WORKSPACE_POOL_PARAMS
+    }
+    pool_to_target_attributes = {
+        workspace_pool_param: _target_attributes_to_pools(
+            getattr(parsed, workspace_pool_param), targets
+        )
+        for workspace_pool_param in WORKSPACE_POOL_TARGET_PARAMS
+    }
+
+    return WorkspaceMemoryPools(
+        [
+            PoolInfo(
```

Comment:
alternatively if this thing complained when a parameters was missing or invalidly named, that would be ok as a way to resolve my earlier comment

Reply:
I agree with this approach, if PoolInfo complained about an incorrect parameter being supplied then TVMC should complain as well

Reply:
I think we still need to keep-in-sync comment because PoolInfo could be added by arguments with defaults and tvmc support for that could be missed out.

Reply:
My concern was PoolInfo could be augmented with new parameters with defaults and we could easily miss those out in the tvmc layer. Having the comment will help the author and the reviewer to not to miss them out in the tvmc layer.

Reply:
Oh we only need the keep-in-sync comment at the PoolInfo side and not here, because "PoolInfo will get upset if you pass incorrect properties".